index.js

```diff
@@ -1,300 +1,147 @@
-  MSG.formats['ssb/gabby-grove'].code,
-])
-const BENDYBT_MSG_TF = Buffer.from([
-  MSG.code,
-  MSG.formats['ssb/bendy-butt'].code,
-])
-
-const BLOB = NAMED_TYPES['blob']
-const BLOB_T = Buffer.from([BLOB.code])
-const CLASSIC_BLOB_TF = Buffer.from([
-  BLOB.code,
-  BLOB.formats['ssb/classic'].code,
-])
-
-const SIGNATURE = NAMED_TYPES['signature']
-const SIGNATURE_T = Buffer.from([SIGNATURE.code])
-const SIGNATURE_TF = Buffer.from([
-  SIGNATURE.code,
-  SIGNATURE.formats['ed25519'].code,
-])
-
-const BOX = NAMED_TYPES['encrypted']
-const BOX_T = Buffer.from([BOX.code])
-const BOX1_TF = Buffer.from([BOX.code, BOX.formats['box1'].code])
-const BOX2_TF = Buffer.from([BOX.code, BOX.formats['box2'].code])
-
-const GENERIC = NAMED_TYPES['generic']
-const STRING_TF = Buffer.from([
-  GENERIC.code,
-  GENERIC.formats['UTF8 string'].code,
-])
-const BOOL_TF = Buffer.from([GENERIC.code, GENERIC.formats['boolean'].code])
-const BOOL_TRUE = Buffer.from([1])
-const BOOL_FALSE = Buffer.from([0])
-const NIL_TF = Buffer.from([GENERIC.code, GENERIC.formats['nil'].code])
-const NIL_TFD = NIL_TF
+const BOOL = {
+  TF: Buffer.from([6, 1]),
+  TRUE: Buffer.from([1]),
+  FALSE: Buffer.from([0]),
+}
+const NIL = {
+  TF: Buffer.from([6, 2]),
+}
+NIL.TFD = NIL.TF
 
 const encoder = {
-  feed(feedId) {
-    let tf
-    if (feedId.endsWith('.ed25519')) tf = CLASSIC_FEED_TF
-    else if (feedId.endsWith('.ggfeed-v1')) tf = GABBYGR_FEED_TF
-    else if (feedId.endsWith('.bbfeed-v1')) tf = BENDYBT_FEED_TF
-    else throw new Error('Unknown feed format: ' + feedId)
-
-    const dotIndex = feedId.lastIndexOf('.')
-    const b64part = feedId.substring(1, dotIndex)
-    const d = Buffer.from(b64part, 'base64')
-
-    return Buffer.concat([tf, d])
+  boolean(input) {
+    const d = input ? BOOL.TRUE : BOOL.FALSE
+    return Buffer.concat([BOOL.TF, d])
   },
+  classic(input, type, format) {
+    let data = input
+    if (type.sigil) data = data.slice(1)
+    if (format.suffix) data = data.slice(0, -format.suffix.length)
 
-  message(msgId) {
-    let tf
-    if (msgId.endsWith('.sha256')) tf = CLASSIC_MSG_TF
-    else if (msgId.endsWith('.ggmsg-v1')) tf = GABBYGR_MSG_TF
-    else if (msgId.endsWith('.bbmsg-v1')) tf = BENDYBT_MSG_TF
-    else throw new Error('Unknown msg ID: ' + msgId)
-
-    const dotIndex = msgId.lastIndexOf('.')
-    const b64part = msgId.substring(1, dotIndex)
-    const d = Buffer.from(b64part, 'base64')
-
-    return Buffer.concat([tf, d])
+    return Buffer.concat([type.code, format.code, Buffer.from(data, 'base64')])
   },
-
-  blob(blobId) {
-    let tf
-    if (blobId.endsWith('.sha256')) tf = CLASSIC_BLOB_TF
-    else throw new Error('Unknown blob ID: ' + blobId)
-
-    const dotIndex = blobId.lastIndexOf('.')
-    const b64part = blobId.substring(1, dotIndex)
-    const d = Buffer.from(b64part, 'base64')
-
-    return Buffer.concat([tf, d])
+  string(input) {
+    return Buffer.concat([Buffer.from([6, 0]), Buffer.from(input, 'utf8')])
```

Comment:
This should be using `STRING.TF`, not the hard-coded bytes

Reply:
Great spotting, this is much better